mono/metadata/icall.c

```diff
@@ -832,36 +853,44 @@ ves_icall_System_Array_FastCopy (MonoArray *source, int source_idx, MonoArray* d
 }
 
 ICALL_EXPORT void
-ves_icall_System_Array_GetGenericValueImpl (MonoArray *arr, guint32 pos, gpointer value)
+ves_icall_System_Array_GetGenericValueImpl (MonoArrayHandle arr, guint32 pos, gpointer value, MonoError* error)
```

Comment:
Wow, we have generic icalls!

I'm not 100% sure what the handle marshaling code will do with `value` - I guess we don't pass a `MonoObjectHandleOut value` argument here, but I'm not sure why.

Reply:
You know/think we have test coverage here? I could write new code to exercise all the paths..

Reply:
I wonder if we shouldn't as part of this, this very PR, expose more very unsafe native stuff, like `mono_gc_memmove_atomic`  and `mono_gc_wbarrier_generic_store` and `mono_gc_bzero_atomic`, accepting void* and UIntPTR  and then do the rest on the managed side. The number of icalls made per scenario would be the same. I don't think perf would suffer, and I suspect cleanliness would increase. You wouldn't have handles any longer in these paths, to worry about the marshaling. Just 'fixed' and void*.

Reply:
Suspecting generic versions should wait until C# has fixed generics in 7.4 or 8.0 depending on timing, or depending on marshaling.

Reply:
Let's do something like dial this back to 3, and/or add test coverage, and/or wait for that other PR.